aws/resource_aws_wafregional_xss_match_set_test.go

```diff
@@ -0,0 +1,235 @@
+	var v waf.XssMatchSet
+	xssMatchSet := fmt.Sprintf("xssMatchSet-%s", acctest.RandString(5))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafRegionalXssMatchSetDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSWafRegionalXssMatchSetConfig(xssMatchSet),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafRegionalXssMatchSetExists("aws_wafregional_xss_match_set.xss_match_set", &v),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_xss_match_set.xss_match_set", "name", xssMatchSet),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_xss_match_set.xss_match_set", "xss_match_tuples.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSWafRegionalXssMatchSet_changeNameForceNew(t *testing.T) {
+	var before, after waf.XssMatchSet
+	xssMatchSet := fmt.Sprintf("xssMatchSet-%s", acctest.RandString(5))
+	xssMatchSetNewName := fmt.Sprintf("xssMatchSetNewName-%s", acctest.RandString(5))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafRegionalXssMatchSetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSWafRegionalXssMatchSetConfig(xssMatchSet),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafRegionalXssMatchSetExists("aws_wafregional_xss_match_set.xss_match_set", &before),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_xss_match_set.xss_match_set", "name", xssMatchSet),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_xss_match_set.xss_match_set", "xss_match_tuples.#", "2"),
+				),
+			},
+			{
+				Config: testAccAWSWafRegionalXssMatchSetConfigChangeName(xssMatchSetNewName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafRegionalXssMatchSetExists("aws_wafregional_xss_match_set.xss_match_set", &after),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_xss_match_set.xss_match_set", "name", xssMatchSetNewName),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_xss_match_set.xss_match_set", "xss_match_tuples.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSWafRegionalXssMatchSet_disappears(t *testing.T) {
+	var v waf.XssMatchSet
+	xssMatchSet := fmt.Sprintf("xssMatchSet-%s", acctest.RandString(5))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafRegionalXssMatchSetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSWafRegionalXssMatchSetConfig(xssMatchSet),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafRegionalXssMatchSetExists("aws_wafregional_xss_match_set.xss_match_set", &v),
+					testAccCheckAWSWafRegionalXssMatchSetDisappears(&v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
```

Comment:
Because there has been a bug in the past affecting all WAF resources I'd like to see 2 more tests here, specifically with no tuples and another one changing tuples, see https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_waf_xss_match_set_test.go#L111-L197 for inspiration and https://github.com/terraform-providers/terraform-provider-aws/commit/99e75ad4b733c8ee66e0d9e911f36e9c7ae2b062 for context (bugfix).